### PR TITLE
Add linear objective support

### PR DESCRIPTION
**Pull request by amontoison**
@dpo

### CODE REVIEW

File: src/moi_nlp_model.jl [hunk 1]
```diff
@@ -148,23 +155,36 @@ end
 # end
 
 function NLPModels.hess_structure!(nlp :: MathOptNLPModel, rows :: AbstractVector{<: Integer}, cols :: AbstractVector{<: Integer})
-  hesslag_struct = MOI.hessian_lagrangian_structure(nlp.eval)
-  for index = 1 : nlp.meta.nnzh
-    rows[index] = hesslag_struct[index][1]
-    cols[index] = hesslag_struct[index][2]
+  if (nlp.eval.has_nlobj) || (nlp.meta.nnln == 0)
```
Comment (dpo) [inline src/moi_nlp_model.jl]: Should it be `nnln > 0`?

**Comment by dpo** [comment]
Would be great to add a test with a linear objective (e.g., hs39) and one with linear constraints only (e.g., hs28).

File: src/moi_nlp_model.jl (hunk 1 above)
Comment (amontoison) [reply src/moi_nlp_model.jl]: good catch, I missed it.

**Comment by dpo** [comment]
Great, thanks!

Strange that we're loosing a bit of coverage [here](https://coveralls.io/builds/29681865/source?filename=src/moi_nlp_model.jl#L165). Aren't these functions called in the consistency check?

@abelsiqueira

**Comment by abelsiqueira** [comment]
Apparently `nlp.eval.has_nlobj = true` for `lincon`. I tested putting a print inside `hess_structure!`.

**Comment by dpo** [comment]
So we need to add an LP to the test problems?

**Comment by abelsiqueira** [comment]
My mistake on the `lincon` part, I mixed objective and constraints in my head. HS10 has linear objective, so we just need to change from `@NLobjective` to `@objective`.

**Comment by dpo** [comment]
HS39 is already like that. Why isn't it activating those lines of code?

**Comment by abelsiqueira** [comment]
`HS39` is not being tested, because it doesn't have a `NLPModels.jl` counterpart.

**Comment by dpo** [comment]
We should remove it if it's not tested. But we need one with linear constraints that will be tested.

**Comment by amontoison** [comment]
Should be good now. We don't need to fill `rows`, `cols` and `vals` with zeros if `nnzh == 0`.

**Comment by dpo** [comment]
Thanks.

**Pull request merged**